app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitTrackChairRatingTypesApiController.php

```diff
@@ -0,0 +1,243 @@
+            },
+            function () {
+                return [];
+            },
+            function ($filter) use ($summit, $selection_plan) {
+                if ($filter instanceof Filter) {
+                    $filter->addFilterCondition(FilterElement::makeEqual('summit_id', $summit->getId()));
+                    $filter->addFilterCondition(FilterElement::makeEqual('selection_plan_id', $selection_plan->getId()));
+                }
+                return $filter;
+            },
+            function () {
+                $current_user = $this->resource_server_context->getCurrentUser();
+                if(!is_null($current_user)){
+                    if(
+                        $current_user->isOnGroup(IGroup::Administrators) ||
+                        $current_user->isOnGroup(IGroup::SuperAdmins) ||
+                        $current_user->isOnGroup(IGroup::TrackChairsAdmins) ||
+                        $current_user->isOnGroup(IGroup::SummitAdministrators)
+                    )
+                        return SerializerRegistry::SerializerType_Private;
+                }
+                return SerializerRegistry::SerializerType_Public;
+            }
+        );
+    }
+
+    /**
+     * @param $summit_id
+     * @param $selection_plan_id
+     * @param $type_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function getTrackChairRatingType($summit_id, $selection_plan_id, $type_id) {
+
+        return $this->processRequest(function () use ($summit_id, $selection_plan_id, $type_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $selection_plan = $this->selection_plan_repository->getById(intval($selection_plan_id));
+            if (is_null($selection_plan)) return $this->error404();
+
+            $track_chair_rating_type = $this->service->getTrackChairRatingType($selection_plan, intval($type_id));
+            if (is_null($track_chair_rating_type)) return $this->error404();
+
+            return $this->ok(SerializerRegistry::getInstance()->getSerializer($track_chair_rating_type)->serialize());
+        });
+    }
+
+    /**
+     * @param $summit_id
+     * @param $selection_plan_id
+     * @return IEntity
+     */
+    public function addTrackChairRatingType($summit_id, $selection_plan_id) {
+
+        return $this->processRequest(function () use ($summit_id, $selection_plan_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $selection_plan = $this->selection_plan_repository->getById(intval($selection_plan_id));
+            if (is_null($selection_plan)) return $this->error404();
+
+            $payload = $this->getJsonPayload($this->getRatingTypeValidationRules());
```

Comment:
pleas use a factory its the pattern that we use for validation rules

Reply:
done